projects/step-frontend/src/lib/modules/timeseries/modules/_common/injectables/dashboard-url-params.service.ts

```diff
@@ -0,0 +1,163 @@
+import { min } from 'rxjs';
+import { TimeSeriesConfig } from '../types/time-series/time-series.config';
+
+const MIN_SUFFIX = '_min';
+const MAX_SUFFIX = '_max';
+
+export interface DashboardUrlParams {
+  // [key: any]: string | undefined;
+  refresh?: string;
+  timeRange?: TimeRangeSelection;
+  grouping?: string[];
+  filters: UrlFilterAttribute[];
+  relativeRange?: string;
+  editMode?: boolean;
+}
+
+export interface UrlFilterAttribute {
+  attribute: string;
+  min?: number;
+  max?: number;
+  values?: string[];
+}
+
+@Injectable()
+export class DashboardUrlParamsService {
+  private _activatedRoute = inject(ActivatedRoute);
+  private _router = inject(Router);
+
+  collectUrlParams(): DashboardUrlParams {
+    let params = this._activatedRoute.snapshot.queryParams;
+    console.log('PARAMS', params);
+    params = Object.keys(params)
+      .filter((key) => key.startsWith(TimeSeriesConfig.DASHBOARD_URL_PARAMS_PREFIX))
+      .reduce((acc, key) => {
+        acc[key.substring(TimeSeriesConfig.DASHBOARD_URL_PARAMS_PREFIX.length)] = params[key];
+        return acc;
+      }, {} as Params);
+    const editModeValue = params['edit'];
+    return {
+      editMode: editModeValue === '1',
+      timeRange: this.extractTimeRange(params),
+      grouping: params['grouping']?.split(','),
+      filters: this.decodeUrlFilters(params),
+    };
+  }
+
+  private extractTimeRange(params: Params): TimeRangeSelection | undefined {
+    const rangeType = params['rangeType'] as TimeRangeType;
+    switch (rangeType) {
+      case TimeRangeType.ABSOLUTE:
+        const from = parseInt(params['from']);
+        const to = parseInt(params['to']);
+        if (!from || !to) {
+          return undefined;
+        }
+        return { type: rangeType, absoluteSelection: { from: from, to: to } };
+      case TimeRangeType.RELATIVE:
+        const relativeRange = parseInt(params['relativeRange']);
+        return { type: rangeType, relativeSelection: { timeInMs: relativeRange } };
+      default:
+        return undefined;
+    }
+  }
+
+  private decodeUrlFilters(params: Params): UrlFilterAttribute[] {
+    return Object.entries(params)
+      .filter(([key]) => key.startsWith(TimeSeriesConfig.DASHBOARD_URL_FILTER_PREFIX))
+      .map(([key, value]) => {
+        key = key.substring(TimeSeriesConfig.DASHBOARD_URL_FILTER_PREFIX.length);
+
+        const filterItem: UrlFilterAttribute = { attribute: key };
+
+        if (key.endsWith(MIN_SUFFIX)) {
+          filterItem.attribute = key.substring(0, -4);
+          filterItem.min = value;
+        } else if (key.endsWith(MAX_SUFFIX)) {
+          filterItem.attribute = key.substring(0, -4);
+          filterItem.max = value;
+        } else {
+          filterItem.values = value.split(',');
+        }
+        return filterItem;
+      })
+      .filter((attr) => attr.values?.length || attr.min != undefined || attr.max != undefined); // not empty filters only
```

Comment:
```suggestion
      .filter((attr) => attr.values?.length || attr.min || attr.max); 
```
(keep the comment if you want but checking for undefined doesn't have to be explicit)